brd-ios/breadwallet/src/Redesign/CLEANerSwift/Controllers/VIPTableViewController.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  File.swift
+//  
+//
+//  Created by Rok Cresnik on 01/12/2021.
+//
+
+import UIKit
+
+
+class VIPTableViewController<C: CoordinatableRoutes,
+                                  I: Interactor,
+                                  P: Presenter,
+                                  DS: BaseDataStore & NSObject>: VIPViewController<C, I, P, DS>,
+                                                                 UITableViewDelegate,
+                                                                 UITableViewDataSource {
+    override var isModalDismissableEnabled: Bool {
+        return true
+    }
+
+    var sections: [AnyHashable] = []
+    var sectionRows: [AnyHashable: [Any]] = [:]
+
+    // MARK: LazyUI
+    lazy var tableView: UITableView = {
+        var tableView = UITableView(frame: .zero, style: .grouped)
+        tableView.delegate = self
+        tableView.dataSource = self
+        // this prevents the top offset on tableViews
+        let zeroView = UIView(frame: .init(origin: .zero, size: .init(width: 0, height: CGFloat.leastNonzeroMagnitude)))
+        tableView.tableHeaderView = zeroView
+        tableView.tableFooterView = zeroView
+        tableView.estimatedSectionHeaderHeight = CGFloat.leastNormalMagnitude
+        tableView.estimatedRowHeight = CGFloat.leastNormalMagnitude
+        tableView.estimatedSectionFooterHeight = CGFloat.leastNormalMagnitude
+
+        tableView.sectionHeaderHeight = UITableView.automaticDimension
+        tableView.rowHeight = UITableView.automaticDimension
+        tableView.sectionFooterHeight = UITableView.automaticDimension
+
+        tableView.separatorStyle = .none
+        return tableView
+    }()
+
+    // MARK: Overrides
+    override func setupSubviews() {
+        super.setupSubviews()
+
+        view.addSubview(tableView)
+        let constraints = [
+            tableView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            tableView.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+            tableView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: view.layoutMargins.left),
```

Comment:
lets never use .left or .right and instead use trailing and leading

Reply:
dont know what you mean.. i am setting leading anchors... and the margins have no leading/trailing